src/v/cloud_storage_clients/client_pool.h

```diff
@@ -128,6 +132,47 @@ class client_pool
     ///         are in use)
     ss::future<client_lease> acquire(ss::abort_source& as);
 
+    /// \brief Acquire http client from the pool for a specified duration.
+    ///
+    /// Same invariants as ::acquire apply (see above).
+    ///
+    /// \param as
+    /// \param to - Lease duration, after which the client is forcibly shut
+    ///             down.
+    /// \param log - Log a message on timeout. The provided logger should have
+    ///              static storage duration since it is captured by a watchdog
+    ///              handler, which will be executed in a background fiber.
+    /// \param ctx - Optional context for the log message. e.g. the string
+    ///              representation of a retry_chain_node.
+    template<typename Logger>
+    ss::future<client_lease> acquire_with_timeout(
+      ss::abort_source& as,
+      ss::lowres_clock::duration to,
+      Logger& log,
+      std::optional<ss::sstring> ctx = std::nullopt) {
+        auto lease = co_await acquire(as);
+        // take a copy of the shared_ptr held by the lease to avoid racing with
+        // client_pool teardown
+        lease._wd.emplace(
```

Comment:
Maybe the deleter of the lease is a better place to allocate the watchdog? It could be one of the captured parameters of the lambda. This way it doesn't have to be moveable. Or maybe this could be a unique_ptr.

Reply:
done w/ `unique_ptr`, which seems much simpler